Sources/General/Commands/Setup.swift

```diff
@@ -0,0 +1,337 @@
+//
+//  Copyright © 2020 Rosberry. All rights reserved.
+//
+
+import Foundation
+import ArgumentParser
+import ZIPFoundation
+
+final class Setup: ParsableCommand {
+
+    enum Error: Swift.Error {
+        case githubName(_ github: String)
+        case url(_ url: String)
+        case download(_ url: URL)
+        case write(_ url: URL)
+        case remove(_ url: URL)
+    }
+
+    static let configuration: CommandConfiguration = .init(commandName: "setup", abstract: "Provides your environment with templates")
+
+    @Option(name: [.customLong("github"), .customShort("g")],
+            default: Constants.defaultTemplatesGithub,
+            help: .init(stringLiteral:
+                "Use this option if templates are placed on github." +
+                " Format: \"<github>\\ [branch]\". Default: \"\(Constants.defaultTemplatesGithub)\""))
+    var githubPath: String
+
+    @Option(name: [.customLong("local"), .customShort("l")],
+            default: false,
```

Comment:
Usually it works the other way around - by default cli tools try to use content from the current directory and fallback to global configuration if no local one is found or if a specific flag (something like `--global` is set).